build.gradle

```diff
@@ -31,8 +31,8 @@ plugins {
 }
 
 // Declare default Java versions for the entire project and for SolrJ separately
-rootProject.ext.minJavaVersionDefault = JavaVersion.VERSION_11
-rootProject.ext.minJavaVersionSolrJ = JavaVersion.VERSION_11
+rootProject.ext.minJavaVersionDefault = JavaVersion.VERSION_21
+rootProject.ext.minJavaVersionSolrJ = JavaVersion.VERSION_21
```

Comment:
We're trying to set different Java versions for SolrJ (17) and Solr (21). However, when I adjust the settings accordingly, the :solr:api test cases start failing because they can't locate :solr:test-framework. I've attached the exception for reference. @dweiss Do you have any pointers on how we can achieve different versions for SolrJ and Solr? Any guidance would be appreciated. Thank you!

```
> Task :solr:api:ecjLintTest FAILED
----------
1. ERROR in /Users/sanjaydutt/Documents/solr/solr/api/src/test/org/apache/solr/util/TestSolrVersion.java (at line 19)
        import org.apache.solr.SolrTestCase;
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
The import org.apache.solr.SolrTestCase cannot be resolved
----------
2. ERROR in /Users/sanjaydutt/Documents/solr/solr/api/src/test/org/apache/solr/util/TestSolrVersion.java (at line 23)
        public class TestSolrVersion extends SolrTestCase {
                                             ^^^^^^^^^^^^
SolrTestCase cannot be resolved to a type

```

Reply:
You'd need to use different toolchains, I think. But I have zero experience with this, sorry.